### PR TITLE
Add star genomegenerate component

### PR DESCRIPTION
**Pull request by sainirmayi**
## Description
Component to generate a genome index for STAR alignment.

## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/star/star_genomegenerate/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      Type of genome transformation
+        None       ... no transformation
+        Haploid    ... replace reference alleles with alternative alleles from VCF file (e.g. consensus allele)
+        Diploid    ... create two haplotypes for each chromosome listed in VCF file, for genotypes 1|2, assumes perfect phasing (e.g. personal genome)
+    example: None
+  - name: --genomeTransformVCF
+    type: file
+    description: path to VCF file for genome transformation
+  
+- name: "Output"
```
Comment (tverbeiren) [inline src/star/star_genomegenerate/config.vsh.yaml:94]: I think it's better to make this a `required: true` argument.

File: src/star/star_genomegenerate/test.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+EOF
+
+cat > genes.gtf <<'EOF'
+chr1    example_source  gene    0    50   .   +   .   gene_id "gene1"; transcript_id "transcript1";
+chr1    example_source  exon    20   40   .   +   .   gene_id "gene1"; transcript_id "transcript1"; 
+EOF
+
+#########################################################################################
+
+mkdir star_index
```
Comment (tverbeiren) [inline src/star/star_genomegenerate/test.sh:26]: This is covered in the component script itself, right?

File: src/star/star_genomegenerate/config.vsh.yaml [hunk 3]
```diff
@@ -0,0 +1,138 @@
+name: star_genomegenerate
```
Comment (tverbeiren) [inline src/star/star_genomegenerate/config.vsh.yaml:1]: I wonder if we should call this just `generate_genome` for instance and add the namespace `star` instead?

TBD

**Review by tverbeiren — CHANGES_REQUESTED** [review]

File: src/star/star_genomegenerate/config.vsh.yaml (hunk 3 above)
Comment (rcannood) [reply src/star/star_genomegenerate/config.vsh.yaml:1]: well, `name: star_genomegenerate` is in line with other components in biobase.

I might consider 

```suggestion
name: star_genome_generate

Comment (tverbeiren) [reply src/star/star_genomegenerate/config.vsh.yaml:1]: If we stick to `start_...` as a conventions, that's fine by me.

In line with `star_align_reads` I would propose `star_generate_genome`.

File: src/star/star_genomegenerate/test.sh (hunk 2 above)
Comment (sainirmayi) [reply src/star/star_genomegenerate/test.sh]: Fixed the script.

File: src/star/star_genomegenerate/config.vsh.yaml (hunk 3 above)
Comment (sainirmayi) [reply src/star/star_genomegenerate/config.vsh.yaml]: Renamed the component.

**Review by rcannood — APPROVED** [review]

**Pull request merged**